### PR TITLE
[Merged by Bors] - malfeasance syncer: count every error as a failed request

### PR DESCRIPTION
**Pull request by poszu**
## Motivation

Treat every fetch error as a failed request in the malfeasance proofs syncer to avoid hanging if the fetcher returns an unrecognized error.

## Test Plan

updated test

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Review by fasmat — APPROVED** [review]

**Review by fasmat — APPROVED** [review]
I thin the `downloadMalfeasanceProofs` is nested too much, for instance the last if could be inverted to:

```go
if len(batch) == 0 {
    log.Debug(...)
    continue
}
```

avoiding the need to put the happy path into a big block inside the `if`. The blocks before that can possibly be optimized for readability as well 🙂

**Comment by poszu** [comment]
> I thin the `downloadMalfeasanceProofs` is nested too much, for instance the last if could be inverted to:
> 
> ```go
> if len(batch) == 0 {
>     log.Debug(...)
>     continue
> }
> ```
> 
> avoiding the need to put the happy path into a big block inside the `if`. The blocks before that can possibly be optimized for readability as well 🙂

The `downloadMalfeasanceProofs` is generally very convoluted and hard to follow :(

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
Flaky TestCheckpoint

bors merge
